### ISSUE TITLE
[FEATURE]: Implement Agent Framework Chat Options Integration

### ISSUE THREAD

**Issue by Josephrp** [huggingface-science/DeepCritical#72]
## Feature Description

Implement integration of agent framework chat options into the DeepAgent system for comprehensive model configuration, tool settings, and request customization.

## Problem Statement

The DeepAgent system needs to properly integrate the vendored agent framework chat options to provide flexible model configuration, tool management, and request customization across all agent interactions.

### 1\. **ChatOptions Integration**

Integrate the `ChatOptions` class from `agent_framework_options.py`:

#### Core Configuration Fields

* **model_id**: Model identifier for provider routing
* **allow_multiple_tool_calls**: Enable multiple tool execution in single response
* **conversation_id**: Track conversation context across requests
* **frequency_penalty**: Control response creativity and repetition
* **instructions**: System-level instructions for agent behavior
* **logit_bias**: Fine-tune token probabilities for specific behaviors
* **max_tokens**: Limit response length
* **metadata**: Custom metadata for tracking and debugging
* **presence_penalty**: Encourage/discourage topic exploration
* **response_format**: Specify expected response structure
* **seed**: Ensure reproducible responses
* **stop**: Define custom stop sequences
* **store**: Enable conversation storage
* **temperature**: Control response randomness
* **tool_choice**: Configure tool usage behavior
* **tools**: List of available tools for the agent
* **top_p**: Nucleus sampling parameter
* **user**: User identifier for personalized responses

#### Advanced Features

* **Tool Choice Modes**: AUTO, REQUIRED, NONE with optional function names
* **Tool Combination**: Merge tool lists from multiple option sources
* **Provider Settings**: Convert options to provider-specific formats
* **Validation**: Comprehensive field validation with proper ranges

### 2\. **ToolMode Integration**

Integrate the `ToolMode` enum for tool execution control:

#### Tool Mode Types

* **AUTO**: Agent decides whether to use tools based on context
* **REQUIRED**: Must use at least one tool (optionally specify which)
* **NONE**: Disable tool usage entirely

#### Implementation Features

* **Dynamic Configuration**: Change tool modes per request
* **Function-Specific Requirements**: Require specific tools by name
* **Provider Compatibility**: Convert modes to provider-specific formats
* **Error Handling**: Proper validation and error messages

### 3\. **Configuration Management**

Implement comprehensive configuration handling:

#### Option Composition

* **Merging Strategy**: Combine options from multiple sources
* **Precedence Rules**: Determine which settings take priority
* **Validation Pipeline**: Validate combined options before use
* **Default Handling**: Provide sensible defaults for missing options

#### Provider Integration

* **Format Conversion**: Transform options to provider-specific formats
* **Compatibility Layer**: Handle differences between providers
* **Fallback Mechanisms**: Graceful degradation for unsupported features
* **Version Handling**: Support for provider API version differences

### ChatOptions Usage

```python
from DeepResearch.src.datatypes.agent_framework_options import ChatOptions, ToolMode

# Basic configuration
options = ChatOptions(
    model_id="anthropic:claude-sonnet-4-0",
    temperature=0.7,
    max_tokens=1000
)

# Tool configuration
tool_options = ChatOptions(
    tool_choice=ToolMode.AUTO,
    tools=[search_tool, file_tool]
)

# Combine options
combined = options & tool_options
```

### Tool Mode Configuration

```python
from DeepResearch.src.datatypes.agent_framework_options import ToolMode

# Auto tool selection
auto_mode = ToolMode.AUTO

# Require specific tool
required_mode = ToolMode.REQUIRED("search_web")

# Disable tools
none_mode = ToolMode.NONE

# Use in chat options
options = ChatOptions(tool_choice=required_mode)
```

### Provider Settings Conversion

```python
class ProviderAdapter:
    def to_anthropic_settings(self, options: ChatOptions) -> Dict[str, Any]:
        # Convert ChatOptions to Anthropic-specific format
        return options.to_provider_settings(by_alias=True)

    def to_openai_settings(self, options: ChatOptions) -> Dict[str, Any]:
        # Convert ChatOptions to OpenAI-specific format
        return options.to_provider_settings(by_alias=False)
```

## Integration Points

* **Agent Configuration**: All agents use ChatOptions for model settings
* **Tool Management**: Tool availability and behavior controlled via ChatOptions
* **Request Processing**: Each request validated against ChatOptions schema
* **Provider Routing**: Options determine which provider and model to use
* **Performance Tuning**: Options control response quality and speed trade-offs

## Testing Requirements

* Unit tests for ChatOptions field validation and conversion
* Integration tests for option composition and merging
* Provider-specific format conversion tests
* Tool mode configuration and execution tests
* Error handling tests for invalid option combinations

## Priority

High - Essential for flexible agent configuration and provider compatibility.

## Parent Issue

DEE-39: \[FEATURE\]: Integrate Agent Framework Types into DeepAgent System
